src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeed.cs

```diff
@@ -46,23 +47,51 @@ public BlobFeed(string accountName, string accountKey, string containerName, str
             string url = $"{FeedContainerUrl}/{blobPath}?comp=metadata";
             using (HttpClient client = new HttpClient())
             {
-                client.DefaultRequestHeaders.Clear();
-                var request = AzureHelper.RequestMessage("GET", url, AccountName, AccountKey).Invoke();
-                using (HttpResponseMessage response = await client.SendAsync(request))
+                const int MaxAttempts = 15;
+                // add a bit of randomness to the retry delay
+                var rng = new Random();
+                int retryCount = MaxAttempts;
+
+                while (true)
                 {
-                    if (response.IsSuccessStatusCode)
+                    try
                     {
-                        Log.LogMessage(
-                            MessageImportance.Low,
-                            $"Blob {blobPath} exists for {AccountName}: Status Code:{response.StatusCode} Status Desc: {await response.Content.ReadAsStringAsync()}");
+                        client.DefaultRequestHeaders.Clear();
+                        var request = AzureHelper.RequestMessage("GET", url, AccountName, AccountKey).Invoke();
+                        using (HttpResponseMessage response = await client.SendAsync(request))
+                        {
+                            if (response.IsSuccessStatusCode)
+                            {
+                                Log.LogMessage(
+                                    MessageImportance.Low,
+                                    $"Blob {blobPath} exists for {AccountName}: Status Code:{response.StatusCode} Status Desc: {await response.Content.ReadAsStringAsync()}");
+                            }
+                            else
+                            {
+                                Log.LogMessage(
+                                    MessageImportance.Low,
+                                    $"Blob {blobPath} does not exist for {AccountName}: Status Code:{response.StatusCode} Status Desc: {await response.Content.ReadAsStringAsync()}");
+                            }
+                            return response.IsSuccessStatusCode;
+                        }
                     }
-                    else
+                    catch (HttpRequestException toLog)
```

Comment:
You came here to catch timeouts;  THose are TaskCancelledExceptions not HttpRequestExceptions.

Reply:
From the issue that sparked this:

Unable to upload to assets/symbols/NETStandard.Library.2.1.0-preview1-26704-01.symbols.nupkg due to System.Net.Http.HttpRequestException: An error occurred while sending the request. ---> System.Net.WebException: Unable to connect to the remote server ---> System.Net.Sockets.SocketException: A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond 13.66.176.16:443

It seems it is being surfaced as an HttpRequestException. Or am I interpreting these wrong?

Reply:
Both can happen, I don't quite know why it decides to do which it does.